frontend/lib/resolution/scope-queries.cpp

```diff
@@ -1006,13 +1006,56 @@ convertOneRename(UniqueString oldName, UniqueString newName) {
   return ret;
 }
 
+static bool
+validateAndPushRename(Context* context,
+                      const VisibilityClause* visibilityClause,
+                      std::vector<std::pair<UniqueString,UniqueString>>& renames,
+                      std::pair<UniqueString, UniqueString> toPush) {
+  auto toPushNode = visibilityClause->limitation(renames.size());
+  // Use an indexed loop so we can use visibilityClause->limitation(i)
+  for (size_t i = 0; i < renames.size(); i++) {
+    auto& rename = renames[i];
+    auto renameNode = visibilityClause->limitation(i);
+
+    if (rename.second == toPush.second) {
+      // Renamed to the same thing, that's an error.
+      CHPL_REPORT(context, UseImportMultiplyDefined,
+                  rename.second, renameNode, toPushNode);
+      return false;
+    }
+
+    if (rename.first == toPush.first) {
+      // Renamed from the same thing. Possibly a mistake, but not fatal.
+      CHPL_REPORT(context, UseImportMultiplyMentioned,
+                  rename.first, renameNode, toPushNode);
+    } else if (rename.second == toPush.first) {
+      // Rename chain like `a as b, b as c`.
+      CHPL_REPORT(context, UseImportTransitiveRename,
+                  rename.first, toPush.first, toPush.second,
+                  renameNode, toPushNode);
+    } else if (rename.first == toPush.second) {
+      // Rename chain like `b as c, a as b`.
+      // Maybe do not warn here?
```

Comment:
is this a TODO? what does the production compiler do here?

Reply:
The production compiler does warn here, I believe, but I think that it shouldn't. The reason I believe so is twofold:

1. The user listed the rename from `b` before they listed the rename _to_ `b`. To me, this is an indication of deliberate intent (if they wanted a transitive rename, why list it in reverse order?)
2. The "chain rename" warning is triggered for legitimate use cases, such as the following:
    ```Chapel
    module M {
        var a = "a";
        var b = "b";
        var c = "c";
    }

    module N {
        import M.{b as c, a as b};
        proc main() {
    
        }
    }
    ```
    There is, as far as I know, no way to silence this warning in production. However, by making the backwards case not warn, we could provide such a silence mechanism for the legitimate use cases.
3. An error would still trigger if `b` isn't in `M`; since Chapel typically forbids use-before-def of variables, it shouldn't be too much of a stretch for the user to suspect that (if they believe transitive renames work) they need to move the `b as c` after `a as b`, at which point they'd get the warning.

Reply:
I suppose I can put these reasons in a comment there instead of a "maybe do not warn?".

Reply:
Yes, I was just asking for the comment to be clearer. Thanks.